inc/uninstall.class.php

```diff
@@ -469,6 +473,70 @@ static function deleteFusionInventoryLink($itemtype, $items_id) {
       }
    }
 
+   /**
+    * Function to remove GLPI Inventory information for an asset
+    *
+    * @param CommonDBTM $item
+    *
+    * @return nothing
+   **/
+   static function deleteGlpiInventoryLink($item) {
+
+      $plug = new Plugin();
+      if ($plug->isActivated('glpiinventory') && function_exists('plugin_pre_item_purge_glpiinventory')) {
+
+         // let glpi-inventory to clean item if needed (agent / collect etc ..)
+         plugin_pre_item_purge_glpiinventory($item);
+      } else {
+         $agent = new Agent();
+         $agent->deleteByCriteria([
+            'itemtype' => $item->getType(),
+            'items_id' => $item->getID(),
+         ]);
+      }
+
+      // purge lock manually because related computer is not purged
+      $lockedfield = new Lockedfield();
+      if ($lockedfield->isHandled($item)) {
+         $lockedfield->itemDeleted();
+      }
+
+      // unlock item relations
+      global $DB;
+      $RELATION = getDbRelations();
+      if (isset($RELATION[$item->getTable()])) {
+         foreach ($RELATION[$item->getTable()] as $tablename => $field) {
+            if ($tablename[0] == '_') {
+               $tablename = ltrim($tablename, '_');
+            }
+            if (is_array($field)) {
+               $DB->update(
+                  $tablename,
+                  [
+                     'is_deleted' => 0
+                  ],
+                  [
+                     $field[0] => $item->getID(),
+                     $field[1] => $item->getType(),
+                     'is_dynamic' => 1
+                  ]
+               );
+            } else {
+               $DB->update(
+                  $tablename,
+                  [
+                     'is_deleted' => 0
+                  ],
+                  [
+                     $field => $item->getID(),
+                     'is_dynamic' => 1
+                  ]
+               );
+            }
```

Comment:
Technically, you may have an array containing 2 distinct foreign keys, for instance
```php
    'glpi_groups' => [
        'glpi_certificates'          => [
            'groups_id_tech',
            'groups_id',
        ],
```
, also, you cannot be sure that `itemtype` and `items_id` are declared in the correct order.


```suggestion
            if (
               is_array($field)
               && count($itemtype_match = preg_grep('/itemtype/', $field)) === 1
               && count($items_id_match = preg_grep('/items_id/', $field)) === 1
            ) {
               $DB->update(
                  $tablename,
                  [
                     'is_deleted' => 0
                  ],
                  [
                     $items_id_match[0] => $item->getID(),
                     $itemtype_match[1] => $item->getType(),
                     'is_dynamic' => 1
                  ]
               );
            } else {
               if (!is_array($field)) {
                  $field = [$field];
               }
               foreach ($field as $f) {
                  $DB->update(
                     $tablename,
                     [
                        'is_deleted' => 0
                     ],
                     [
                        $f => $item->getID(),
                        'is_dynamic' => 1
                     ]
                  );
               }
            }
```